### PR TITLE
Issue #1108 - Django 4.2.3 dependabot issue

### LINKED ISSUE #1108: Fix Broken dependabot bump- django 4.2.3
### Issue Description

Troubleshoot and fix the update that is currently failing going from 4.2.1 to 4.2.3 of Django. A minor change shouldn't be resulting in errors, but this one causes failures. 

see branch dependabot/pip/src/django-4.2.3 titled Bump django from 4.2.1 to 4.2.3 in /src 

### Additional Context (optional)

_No response_

### Issue Links

_No response_

### PR DESCRIPTION
**Pull request by CocoByte**
## Ticket 1108

Resolves #1108

## Changes

- updated Django to 4.2.10
- (also made the necessary manual updates to geventconnpool item in both requirements.txt and PipFile.lock)

## Context for reviewers

Looks like we are good to go with the dependabot issue (unable to replicate error and peer-programmed with Alysia to verify our direction).  We opted to update to Django 4.2.10 (the latest 4.2 version before Django 5)

#### Satisfied acceptance criteria and met development standards

- [X] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [X] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [X] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: src/requirements.txt [hunk 1]
```diff
@@ -1,61 +1,61 @@
...
-faker==20.1.0; python_version >= '3.8'
+environs[django]==10.3.0; python_version >= '3.8'
+faker==23.1.0; python_version >= '3.8'
 fred-epplib@ git+https://github.com/cisagov/epplib.git@d56d183f1664f34c40ca9716a3a9a345f0ef561c
 furl==2.1.3
 future==0.18.3; python_version >= '2.6' and python_version not in '3.0, 3.1, 3.2, 3.3'
 gevent==23.9.1; python_version >= '3.8'
-geventconnpool@ git+https://github.com/rasky/geventconnpool.git@1bbb93a714a331a069adf27265fe582d9ba7ecd4
-greenlet==3.0.1; python_version >= '3.7'
+geventconnpool@ git+https://github.com/rasky/geventconnpool.git
```
Comment (abroddrick) [inline src/requirements.txt:31]: needs to be reverted to what it is on main

Comment (CocoByte) [reply src/requirements.txt]: done

File: src/Pipfile [hunk 2]
```diff
@@ -4,7 +4,7 @@ verify_ssl = true
 name = "pypi"
 
 [packages]
-django = "*"
+django = "4.2.10"
```
[the remark this answers is not in the document]
Comment (CocoByte) [reply src/Pipfile:7]: Yes, and we want that because we don't want to go to version 5 just yet

File: src/requirements.txt [hunk 3]
```diff
@@ -1,61 +1,61 @@
...
-boto3==1.33.7; python_version >= '3.7'
-botocore==1.33.7; python_version >= '3.7'
+boto3==1.34.37; python_version >= '3.8'
+botocore==1.34.37; python_version >= '3.8'
 cachetools==5.3.2; python_version >= '3.7'
-certifi==2023.11.17; python_version >= '3.6'
+certifi==2024.2.2; python_version >= '3.6'
 cfenv==0.5.3
-cffi==1.16.0; python_version >= '3.8'
+cffi==1.16.0; platform_python_implementation != 'PyPy'
```
Comment (zandercymatics) [inline src/requirements.txt:9]: (Q) Just out of curiosity, why was this causing issues? Or was this auto updated?

File: src/requirements.txt [hunk 4]
```diff
@@ -1,61 +1,61 @@
...
-faker==20.1.0; python_version >= '3.8'
+environs[django]==10.3.0; python_version >= '3.8'
+faker==23.1.0; python_version >= '3.8'
 fred-epplib@ git+https://github.com/cisagov/epplib.git@d56d183f1664f34c40ca9716a3a9a345f0ef561c
 furl==2.1.3
 future==0.18.3; python_version >= '2.6' and python_version not in '3.0, 3.1, 3.2, 3.3'
 gevent==23.9.1; python_version >= '3.8'
 geventconnpool@ git+https://github.com/rasky/geventconnpool.git@1bbb93a714a331a069adf27265fe582d9ba7ecd4
-greenlet==3.0.1; python_version >= '3.7'
+greenlet==3.0.3; python_version >= '3.7'
```
Comment (zandercymatics) [inline src/requirements.txt:32]: Nice!

File: src/registrar/views/utility/mixins.py [hunk 5]
```diff
@@ -347,7 +342,6 @@ def has_permission(self):
 
 
 class ApplicationWizardPermission(PermissionsLoginMixin):
-
```
Comment (zandercymatics) [inline src/registrar/views/utility/mixins.py:350]: (Q) Are these intentional for consistency reasons?

File: src/requirements.txt (hunk 3 above)
Comment (CocoByte) [reply src/requirements.txt:9]: This was auto-updated

File: src/registrar/views/utility/mixins.py (hunk 5 above)
Comment (CocoByte) [reply src/registrar/views/utility/mixins.py:350]: This was done by the linter.  Not entirely sure why it is suddenly making changes to all these files...

**Review by zandercymatics — APPROVED** [review]
LGTM

**Pull request merged**